pkg/cache/queue/manager.go

```diff
@@ -584,7 +601,12 @@ func (m *Manager) QueueAssociatedInadmissibleWorkloadsAfter(ctx context.Context,
 		action()
 	}
 
-	q := m.localQueues[queue.KeyFromWorkload(w)]
+	wlKey := workload.Key(w)
+	qKey, ok := m.assignedWorkloads[wlKey]
+	if !ok {
+		return
+	}
```

Comment:
Why do we need this? Is there some scenario where the Workload object doesn't store the valid name of the LQ and the mapping from `assignedWorkloads` does? If so then I'd appreciate if you added a comment documenting that

Reply:
If the signature will be changed in a follow-up PR then I'd like this also to be changed in the same follow-up